modules/internal/ChapelDistribution.chpl

```diff
@@ -798,7 +819,10 @@ module ChapelDistribution {
   // that arr.eltType is meaningful.
   proc _delete_arr(arr, param privatized:bool) {
     // decide whether or not the array is an alias
-    var isalias = (arr._arrAlias != nil);
+    var isalias = ((arr._arrAlias != nil) || 
+                   arr.isSliceArrayView() || 
+                   arr.isRankChangeArrayView() ||
+                   arr.isReindexArrayView());
```

Comment:
You couldn't call 'chpl__isArrayView' ? Is this an order-of-resolution issue or something?